src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/GatewayIPAddressInformationCollection.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.Net.NetworkInformation
+{
+    public class GatewayIPAddressInformationCollection : ICollection<GatewayIPAddressInformation>
+    {
+        private Collection<GatewayIPAddressInformation> _addresses = new Collection<GatewayIPAddressInformation>();
+
+        protected internal GatewayIPAddressInformationCollection()
+        {
+        }
+
+        public virtual void CopyTo(GatewayIPAddressInformation[] array, int offset)
+        {
+            _addresses.CopyTo(array, offset);
+        }
+
+        public virtual int Count
+        {
+            get
+            {
+                return _addresses.Count;
+            }
+        }
+
+        public virtual bool IsReadOnly
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public virtual GatewayIPAddressInformation this[int index]
+        {
+            get
+            {
+                return (GatewayIPAddressInformation)_addresses[index];
```

Comment:
There are multiple unnecessary casts that got here from the desktop code. I'll try to remove all.